go/worker/common/p2p/dispatch.go

```diff
@@ -236,6 +245,57 @@ func (h *topicHandler) retryWorker(m *queuedMsg) {
 	}
 }
 
+func (h *topicHandler) tryPublishing(rawMsg []byte) error {
+	peers := len(h.topic.ListPeers())
+	if peers == 0 {
+		// On init, if there are no peers, the library will sometimes just
+		// swallow the message and mark it as seen without retrying or returning
+		// an error. This special case is to try to preempt that.
+		h.logger.Debug("no connected peers, handing off to retry worker")
+		select {
+		case h.rawMsgQueue <- &rawMessage{rawMsg}:
+			return nil
+		default:
+			return fmt.Errorf("worker/common/p2p: message queue overflow, libp2p still not initialized")
+		}
+	}
+
+	return h.topic.Publish(h.ctx, rawMsg)
+}
+
+func (h *topicHandler) publishWorker() {
```

Comment:
If you rename the channel name, then this could also be:

```suggestion
func (h *topicHandler) pendingMessagesWorker() {
```

With a docstring that this is handles messages when there's no connected peers.